src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -456,13 +488,44 @@ template class LogicalExpression<LogicalOperator::OR>;
 
 namespace detail {
 
+//______________________________________________________________________________
+template <CompOp comparison>
+static std::unique_ptr<PrefilterExpression> makeMirroredExpression(
+    const IdOrLocalVocabEntry& referenceValue) {
+  using enum CompOp;
+  switch (comparison) {
+    case LT:
+      // Id < ?var -> ?var > Id
+      return std::make_unique<GreaterThanExpression>(referenceValue);
+    case LE:
+      // Id <= ?var -> ?var >= Id
+      return std::make_unique<GreaterEqualExpression>(referenceValue);
+    case GE:
+      // Id >= ?var -> ?var <= Id
+      return std::make_unique<LessEqualExpression>(referenceValue);
+    case GT:
+      // Id > ?var -> ?var < Id
+      return std::make_unique<LessThanExpression>(referenceValue);
+    case EQ:
+    case NE:
+      // EQ(==) or NE(!=)
+      // Given that these two relations are symmetric w.r.t. ?var and Id,
+      // no swap regarding the relational operator is necessary.
+      return std::make_unique<RelationalExpression<comparison>>(referenceValue);
+    default:
+      // Unchecked / new valueIdComparators::Comparison case.
+      AD_FAIL();
+  }
+}
+
 //______________________________________________________________________________
 void checkPropertiesForPrefilterConstruction(
     const std::vector<PrefilterExprVariablePair>& vec) {
   auto viewVariable = vec | std::views::values;
   if (!std::ranges::is_sorted(viewVariable, std::less<>{})) {
     throw std::runtime_error(
-        "The vector must contain the <PrefilterExpression, Variable> pairs in "
+        "The vector must contain the <PrefilterExpression, Variable> pairs "
+        "in "
         "sorted order w.r.t. Variable value.");
```

Comment:
Please reformat (easiest : whole comment in one line, and then let clang-format do its thing).